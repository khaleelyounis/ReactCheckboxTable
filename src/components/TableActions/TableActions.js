import React from 'react';
import PropTypes from 'prop-types';
import download from '../../assets/download.png';
import styles from './styles.module.css';

const TableActions = ({ selectedCount }) => {
  return (
    <div className={styles.tableActions} data-testid="tableActions">
      <div
        className={styles.tableActionsCheckbox}
        data-testid="tableActionsCheckbox"
      >
        <input type="checkbox" name="checkbox" data-testid="checkbox" />{' '}
      </div>
      <div
        className={styles.tableActionsSelected}
        data-testid="tableActionsSelected"
      >
        {selectedCount > 0 ? `Selected ${selectedCount}` : 'None Selected'}
      </div>
      <div
        className={styles.tableActionsDownload}
        data-testid="tableActionsDownload"
      >
        <img
          src={download}
          className={styles.downloadIcon}
          data-testid="downloadIcon"
          alt="download icon"
        />
        Download Selected
      </div>
    </div>
  );
};

TableActions.defaultProps = {
  selectedCount: 0,
};

TableActions.propTypes = {
  selectedCount: PropTypes.number.isRequired,
};

export default TableActions;
