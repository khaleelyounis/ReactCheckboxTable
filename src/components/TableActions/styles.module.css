.tableActions {
  display: flex;
  flex: 1;
  flex-direction: row;
  justify-content: flex-start;
}

.tableActions > * {
  margin-right: 2rem;
  transition: margin 0.3s;
}

.tableActionsCheckbox {
  padding: 1rem 1rem 1rem 0.7rem;
}

.tableActionsSelected,
.tableActionsDownload {
  display: flex;
  align-items: center;
  font-size: 20px;
}

.tableActionsDownload:hover {
  cursor: pointer;
}

.tableActionsSelected {
  margin-right: 5rem;
}

.downloadIcon {
  width: 20px;
  height: 20px;
  padding-right: 10px;
}

@media only screen and (max-width: 800px) {
  .tableActions > * {
    margin-right: 1rem;
  }
}
