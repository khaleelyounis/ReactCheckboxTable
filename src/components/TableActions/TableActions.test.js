import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import TableActions from './TableActions';

TableActions.defaultProps = {
  selectedCount: 0,
  checked: false,
  downloadCallback: jest.fn(),
  toggleAll: jest.fn(),
};

describe('TableActions Snapshots', () => {
  it('Should render with checkbox, selected, and download elements', () => {
    const component = render(<TableActions />);
    const checkbox = component.getByTestId('tableActionsCheckbox');
    const selected = component.getByTestId('tableActionsSelected');
    const download = component.getByTestId('tableActionsDownload');

    expect(checkbox).toBeTruthy();
    expect(selected).toBeTruthy();
    expect(download).toBeTruthy();
    expect(component).toMatchSnapshot();
  });

  it('Should have None Selected Text with 0 selected count', () => {
    const component = render(<TableActions />);
    const selected = component.getByTestId('tableActionsSelected');

    expect(selected.textContent).toBe('None Selected');
  });

  it('Should have Selected 2 when selectedCount is 2', () => {
    const component = render(<TableActions selectedCount={2} />);
    const selected = component.getByTestId('tableActionsSelected');

    expect(selected.textContent).toBe('Selected 2');
  });

  it('Should have unchecked checkbox when checked is false', () => {
    const component = render(<TableActions />);
    const checkbox = component.getByTestId('checkbox');

    expect(checkbox.checked).toBe(false);
  });

  it('Should have unchecked checkbox when checked is false', () => {
    const component = render(<TableActions checked />);
    const checkbox = component.getByTestId('checkbox');

    expect(checkbox.checked).toBe(true);
  });
});

describe('TableActions Actions', () => {
  it('Should fire downloadCallback function when download element is clicked', () => {
    const component = render(<TableActions />);
    const download = component.getByTestId('tableActionsDownload');

    fireEvent.click(download);
    expect(TableActions.defaultProps.downloadCallback).toHaveBeenCalled();
  });

  it('Should fire toggleAll function when checkbox is clicked', () => {
    const component = render(<TableActions />);
    const checkbox = component.getByTestId('checkbox');

    fireEvent.click(checkbox);
    expect(TableActions.defaultProps.toggleAll).toHaveBeenCalled();
  });
});
