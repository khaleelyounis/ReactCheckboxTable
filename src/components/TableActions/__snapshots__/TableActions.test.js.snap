// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableActions Snapshots Should render with checkbox, selected, and download elements 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="tableActions"
        data-testid="tableActions"
      >
        <div
          class="tableActionsCheckbox"
          data-testid="tableActionsCheckbox"
        >
          <input
            data-testid="checkbox"
            name="checkbox"
            type="checkbox"
          />
           
        </div>
        <div
          class="tableActionsSelected"
          data-testid="tableActionsSelected"
        >
          None Selected
        </div>
        <div
          class="tableActionsDownload"
          data-testid="tableActionsDownload"
        >
          <img
            alt="download icon"
            class="downloadIcon"
            data-testid="downloadIcon"
            src="download.png"
          />
          Download Selected
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="tableActions"
      data-testid="tableActions"
    >
      <div
        class="tableActionsCheckbox"
        data-testid="tableActionsCheckbox"
      >
        <input
          data-testid="checkbox"
          name="checkbox"
          type="checkbox"
        />
         
      </div>
      <div
        class="tableActionsSelected"
        data-testid="tableActionsSelected"
      >
        None Selected
      </div>
      <div
        class="tableActionsDownload"
        data-testid="tableActionsDownload"
      >
        <img
          alt="download icon"
          class="downloadIcon"
          data-testid="downloadIcon"
          src="download.png"
        />
        Download Selected
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
